core/primer.md

```diff
@@ -2,6 +2,185 @@
+structure. This hinders the interoperability of event brokers CloudEvents had in
+mind. xRegistry therefore comes in with a similar motivation to CloudEvents:
+Harmonize another piece of technology in the messaging / eventing ecosystem to
+increase interoperability and decouple event handling from broker products and
+protocols.
+
+Even though xRegistry was built with eventing in mind, the concept of this
+registry specification is far more powerful. Take a look at the [Possible Use
+Cases](#possible-use-cases) for examples outside the eventing world.
+
+## Design Goals
+
+- **Interoperability:** Enable effortless discovery, validation and versioning
+  of resources.
+- **Standardization:** Establish a high level structure for managing resource
+  metadata behind which vendors can unite.
+- **Extensibility:** Accommodate future innovations by allowing for the
+  definition of custom attributes and extensions tailored to specific needs
+  – not only for the happy path, but errors as well.
+- **Flexibility:** Support various resource types, protocols (MQTT, AMQP, Kafka,
+  NATS, HTTP..) and schema languages (JSON Schema, Avro Schema, Protobuf..)
+  – not just CloudEvents.
+- **Simplicity:** Allow simple use cases that do not need versioning of
+  resources or a full blown schema registry on the server.
+
+### Non-Goals
+
+- **Authentication and Authorization:** Rely on established security standards
+  depending on the registry representation.
+- **Relationships between event channels:** Focus on precisely describing a
+  single event channel before standardizing the relationships between event
+  channels.
+
+## Representations
+
+An xRegistry can be represented in the different ways: a JSON file, a static
+file server and an API server. When building up this spec the API server had the
+highest priority, followed by the file representation and the static file server
+as an option to transition between the first two. To enable this transition, one
+of the design goals is the symmetry between all representations.
+
+Whenever registries get large, you will realistically want to use the API server
+or CLI to create the registry and then use its export function to generate files
+for the file or static file server representation.
+
+### File
+
+The registry is represented as a single JSON file.
+
+The primary use case for this representation is a registry which purpose is to
+describe a single project. It basically becomes a declaration manifest of the
+application and can therefore be used in public git repositories or your own
+projects. If you are new to xRegistry and are just trying to get the idea of the
+project, this is a great point to start.
+
+Writing up a registry by hand can be a great way of getting the idea, but it can
+get tedious quickly. This is when you want to use the API server.
+
+### Static File Server
+
+The registry is represented by multiple JSON files served via a static file
```

Comment:
perhaps: ... by multiple JSON files, where each one represents a single entity within the registry, served via a...